torch/csrc/jit/passes/quantization.cpp

```diff
@@ -676,5 +676,246 @@ TORCH_API script::Module InsertObservers(
   return input_module;
 }
 
+Node* insertQuantDeQuantCall(Value* v, const IValue& qparams, at::ScalarType t, bool insert_after=true) {
+  Graph* g = v->node()->owningGraph();
+  Node* quant = createQuantNode(v, g);
+  Node* intrepr = createIntReprNode(v, g);
+  Node* dequant = createDeQuantNode(v, g);
+  Node* insert_point = insert_after ? v->node() : *g->nodes().begin();
+  WithCurrentScope scope_guard(
+      *insert_point->owningGraph(), insert_point->scope());
+  WithInsertPoint ins(insert_point);
```

Comment:
There is no benefit in using `WithInsertPoint` if we only use explicit `insertBefore`/`insertAfter` calls.

Ideally, we would set the insert point at the callsite of this function and change this function to use `insertNode` in this function. This way we get rid of the `insert_after` argument of this function and make the function branchless.

Reply:
hmm, we still have insertBefore and insertAfter right

Reply:
oh, you mean pass different nodes so that it is always insertAfter/insertBefore?

Reply:
You don't need to pass any nodes, that's the beauty of `WithInsertPoint` - when you create an object of this type (e.g. `WithInsertPoint x (my_insert_point);`) it sets the global insert-point to `my_insert_point`. This global insert point is used by all `Graph::insert*` methods except `insertBefore` and `insertAfter`. When `WithInsertPoint`objects comes out of the scope, it restores the previous insert point.

So, in this case we could just create an object `WithInsertPoint` before calling this function and then this function will just insert its stuff into the right place.

Yes, you would need to change the insertion logic in function to be like `insertBefore`, but you would call `insertNode` instead (`insertNode` ~ `insertBefore(insert_point)`).

Reply:
oh, you mean it will do insertBefore/insertAfter depending on where the node is?

Reply:
doesn't seem to work, I'll land like this and we can talk about how to refactor later.